var/spack/repos/builtin/packages/bison/package.py

```diff
@@ -35,7 +35,8 @@ class Bison(AutotoolsPackage):
 
     version('3.0.4', 'a586e11cd4aff49c3ff6d3b6a4c9ccf8')
 
-    depends_on('m4', type='build')
+    # This is a run-time dependency as well
+    depends_on('m4')
```

Comment:
I would leave these changes out of this PR if you're going to submit them in another PR. Otherwise both PRs will be held up until we decide on what to do about this dependency, and the discussion will be split in multiple places.